Rebus.AdoNet.Tests/SagaPersisterTests.cs

```diff
@@ -318,31 +399,127 @@ public void WhenIgnoringNullProperties_DoesNotSaveNullPropertiesOnInsert()
 		[Test]
 		public void CanCreateSagaTablesAutomatically()
 		{
-			// arrange
+			var tableNames = GetTableNames();
+			Assert.That(tableNames, Contains.Item(SagaTableName), "Missing main saga table?!");
 
-			// act
-			CreatePersister().EnsureTablesAreCreated();
+			var schema = GetColumnSchemaFor(SagaTableName); //< (item1: COLUMN_NAME, item2: DATA_TYPE, [...?])
+			var indexes = GetIndexesFor(SagaTableName);
+			Assert.Multiple(() =>
+			{
+				Assert.That(Dialect.GetDbTypeFor(schema.First(x => x.Item1 == "id").Item2), Is.EqualTo(DbType.Guid), "#0.0");
+				Assert.That(Dialect.GetDbTypeFor(schema.First(x => x.Item1 == "revision").Item2), Is.EqualTo(DbType.Int32), "#0.1");
+				Assert.That(schema.First(x => x.Item1 == "saga_type").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#0.2");
+				Assert.That(schema.First(x => x.Item1 == "data").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#0.3");
+				Assert.That(indexes, Contains.Item($"ix_{SagaTableName}_id_saga_type"), "#0.4");
+			});
+
+			if (_persister is AdoNetSagaPersisterLegacy) //< NOTE: DATA_TYPE for PostgresSQL arrays are _text instead of text[].
+			{
+				Assert.Multiple(() =>
+				{
+					var sagaIndexSchema = GetColumnSchemaFor(SagaIndexTableName);
+					var sagaIndexIndexes = GetIndexesFor(SagaIndexTableName);
+					var isArray = _features.HasFlag(Feature.Arrays);
+					Assert.That(tableNames, Contains.Item(SagaIndexTableName), "Missing saga indexes table?!");
+					Assert.That(Dialect.GetDbTypeFor(sagaIndexSchema.First(x => x.Item1 == "saga_id").Item2), Is.EqualTo(DbType.Guid), "#1.0");
+					Assert.That(sagaIndexSchema.First(x => x.Item1 == "key").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#1.1");
+					Assert.That(sagaIndexSchema.First(x => x.Item1 == "value").Item2, Is.EqualTo("varchar").Or.EqualTo("text"), "#1.2");
+					Assert.That(sagaIndexSchema.First(x => x.Item1 == "values").Item2, Is.EqualTo(isArray ? "varchar[]" : "varchar").Or.EqualTo(isArray ? "_text" : "text"), "#1.3");
+					Assert.That(sagaIndexIndexes, Contains.Item($"ix_{SagaIndexTableName}_key_value"), "#1.4");
+					Assert.That(sagaIndexIndexes, Contains.Item($"ix_{SagaIndexTableName}_key_values"), "#1.5");
+				});
+			}
+			else if (_persister is AdoNetSagaPersisterAdvanced)
+			{
+				Assert.Multiple(() =>
+				{
+					var correlationsType = schema.First(x => x.Item1 == "correlations").Item2;
+					Assert.That(tableNames, Does.Not.Contains(SagaIndexTableName), "Have we created saga indexes table?!");
+					Assert.That(Dialect.GetDbTypeFor(correlationsType), Is.EqualTo(DbType.Object), "#2.0");
+					Assert.That(correlationsType, Is.EqualTo("jsonb"), "#2.1");
 
-			// assert
-			var existingTables = GetTableNames();
-			Assert.That(existingTables, Contains.Item(SagaIndexTableName));
-			Assert.That(existingTables, Contains.Item(SagaTableName));
+					if (Dialect.SupportsGinIndexes)
```

Comment:
This should not depend on Dialect supporting X.. but on this actual test run being setup for X features or use-cases.

Reply:
Some asserts run against SQLdialects which may not support N indexes.. and there isn't a feature flag for this.. so:
1. I may keep this.
2. I can add more flags looking for this properties ie.  if (Dialect.Sup..) _features |= Flag.SupportGinIndexes (?)

Reply:
A Dialect indicates what an db+version supports. However we are testing an specific persister with some specific features enabled.
So, if we are using the Advanced persister, we should ensure a GIN index was created.. this should not depend on asking what supports the Dialect (which btw may be returning false, thus being an error), instead we should ensure that if running tests for Advanced (JSON) persister, a GIN index was created.. And yes, we can opt-out some tests (or asserts) for sqlite [or other cases not supporting GIN], but *still* perform such verifications (ie. GIN was created) as general case.